tests/suite/graceful_recovery_test.go

```diff
@@ -100,8 +102,126 @@ var _ = Describe("Graceful Recovery test", Ordered, Label("functional", "gracefu
 		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
 		runRecoveryTest(teaURL, coffeeURL, ngfPodName, nginxContainerName, files, &ns)
 	})
+
+	It("recovers when drained node is restarted", func() {
+		runRestartNodeTest(teaURL, coffeeURL, files, &ns, true)
+	})
+
+	It("recovers when node is restarted abruptly", func() {
+		// FIXME(bjee19) remove Skip() when https://github.com/nginxinc/nginx-gateway-fabric/issues/1108 is completed.
+		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
+		runRestartNodeTest(teaURL, coffeeURL, files, &ns, false)
+	})
 })
 
+func runRestartNodeTest(teaURL, coffeeURL string, files []string, ns *core.Namespace, drain bool) {
+	nodeNames, err := getNodeNames()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(nodeNames).To(HaveLen(1))
+
+	kindNodeName := nodeNames[0]
+
+	if portFwdPort != 0 {
+		close(portForwardStopCh)
+	}
+
+	if drain {
+		_, err := exec.Command(
+			"kubectl",
+			"drain",
+			kindNodeName,
+			"--ignore-daemonsets",
+			"--delete-local-data",
+		).CombinedOutput()
+		if err != nil {
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		_, err = exec.Command(
+			"kubectl",
+			"delete",
+			"node",
+			kindNodeName,
+		).CombinedOutput()
+		if err != nil {
+			Expect(err).ToNot(HaveOccurred())
+		}
+	}
+
+	containerOutput, err := exec.Command(
```

Comment:
```
CONTAINER ID   IMAGE                           COMMAND                  CREATED              STATUS              PORTS                       NAMES
ac9b28713c4b   envoyproxy/envoy:v1.30.1        "/docker-entrypoint.…"   About a minute ago   Up About a minute   10000/tcp                   kindccm-QANXYI4QIRPA5BSC6GNMUJEG5TEYDYEF5UW7WXIH
ba12feefe758   kindest/node:v1.25.16           "/usr/local/bin/entr…"   3 minutes ago        Up 3 minutes        127.0.0.1:36533->6443/tcp   9553241388-control-plane
15996fc54f   moby/buildkit:buildx-stable-1   "buildkitd --allow-i…"   4 minutes ago        Up 4 minutes                                    buildx_buildkit_builder-498d4307-ad1d-49ad-9779-f1ae8e6f94dc0
```

Is an example output. We would parse this for the kind container name where NGF is running on which is `9553241388-control-plane`.

Reply:
This part of the code has been refactored and I'm unsure if the question is relevant anymore, let me know if it applies to some place else.